lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -95,15 +90,18 @@
 
     this.handlerContainer.setTransport(this.transport);
 
-    this.myIdentifier = new SocketRemoteIdentifier((InetSocketAddress)this.transport.getLocalAddress());
+    InetSocketAddress address = new InetSocketAddress(
+        localAddressProvider.getLocalAddress(),
+        this.transport.getListeningPort());
+    this.myIdentifier = new SocketRemoteIdentifier(address);
```

Comment:
Why is this change necessary?

Reply:
The transport.getLocalAddress() returns the socket connection that refers to the 'hostname' which for containers is invalid since the myIdentifier is passed to the shim.conf and used the evaluator to communicate with the driver.

This change address the issue for containers. Since the localAddressProvider defaults to HostBasedLocalAddressProvider, this change does not break the functionality when containers is not used since in that case, the earlier and new value of myIdentifier refer to the same socket.

---
In reply to: [201101479](https://github.com/sharathmalladi/reef/pull/5#discussion_r201101479) [](ancestors = 201101479)